WordPress/gutenberg
# Post excerpt: Fix JavaScript error and other misc. bug fixes

## What?

Attempts to address feedback about the post excerpt block:

- Use `…` instead of `...`
- The ellipsis shows even if the text is not truncated

Removes one instance of `setExcerpt`. The excerpt does not need to be updated when the length is adjusted. This was a left over from a previous implementation.
This fix assures that there is only one undo step needed when changing the value in the excerpt length setting.

Solves a bug where the excerpt was editable but not saved, when the block is used with a custom post type that does not support excerpts.
Solves a JavaScript error in the editor when the block is used with a custom post type that does not support excerpts.
Introduced in https://github.com/WordPress/gutenberg/pull/44964:

```
edit.js:145 Uncaught TypeError: Cannot read properties of undefined (reading 'trim')
    at PostExcerptEditor (edit.js:145:5)
```
## Why?


## How?


## Testing Instructions
1. Create a new post.
2. Add a post excerpt block.
3. Add less than 10 words to the excerpt.
4. Set the excerpt length setting in the block settings sidebar to 10.
5. Deselect the block: no `… ` should show after the text (this is the fix).
6. Select the block add text so that there are more than 10 words.
7. Deselect the block. the `…` symbol should show.


1. Create a new page.
2. Add more than ten words to the page content.
3. Add a post excerpt block.
4. Select the post excerpt block and confirm that the text can be edited.
5. Set the excerpt length setting in the block settings sidebar to 10.
6. Confirm that the length is applied to the excerpt.

Register a custom post type that does not have support for excerpts.
See https://developer.wordpress.org/reference/functions/register_post_type/ for sample code

1. Create a new post item belonging to the custom post type.
2. Add some content.
3. Add a post excerpt block.
4. Select the post excerpt block and confirm that the text can **not** be edited.
5. Save and view the front. An  autogenerated excerpt, matching the length of the post excerpt length option of the block, should show on the front. This matches the behavior of the `get_the_excerpt` function when it is used with a custom post type without excerpt support, without the block.
